integration-test/test.sh

```diff
@@ -0,0 +1,350 @@
+#!/bin/bash
+set -eou pipefail
+IFS=$'\n\t'
+
+# these flighting values should match Makefile
+readonly TEST_SUBS="c3dfd792-49a4-4b06-80fc-6fc6d06c4742"
+readonly TEST_REGION="Brazil South"
+
+# supported images (add/update them as new major versions come out)
+readonly DISTROS=(
+	"2b171e93f07c4903bcad35bda10acf22__CoreOS-Beta-877.1.0" \
+	"5112500ae3b842c8b9c604889f8753c3__OpenLogic-CentOS-71-20150731" \
+	"b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-15_04-amd64-server-20151201-en-us-30GB" \
+	"b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04_3-LTS-amd64-server-20151117-en-us-30GB" )
+
+# Test constants
+readonly SCRIPT_DIR=$(dirname $0)
+readonly CONCURRENCY=10
+readonly DOCKER_CERTS_DIR=dockercerts
+readonly VM_PREFIX=dockerexttest-
+readonly VM_USER=azureuser
+readonly EXTENSION_NAME=DockerExtension
+readonly EXTENSION_PUBLISHER=Microsoft.Azure.Extensions
+readonly EXTENSION_VERSION=1.0
+readonly EXTENSION_CONFIG=extensionconfig/public.json
+readonly EXTENSION_CONFIG_PROT=extensionconfig/protected.json
+expected_extension_version=
+
+
+### Functions
+
+log() {
+	echo "[$(date +%T)][DEBUG]" "$@"
+}
+
+err() {
+	echo >&2 "[$(date +%T)][ERROR]" "$@"
+}
+
+is_empty() {
+    local var=$1
+    [[ -z $var ]]
+}
+
+is_file() {
+    local file=$1
+    [[ -f $file ]]
+}
+
+
+command_exists() {
+	command -v "$@" > /dev/null 2>&1
+}
+
+check_deps() {
+	local deps=(azure jq docker curl parallel)
+
+	local cmd=
+	for cmd in "${deps[@]}"; do
+		command_exists $cmd || { err "$cmd not installed."; exit 1; }
+	done
+}
+
+check_asm() {
+	# capture "Current Mode: arm" from azure cmd output
+	if [[ "$(azure)" != *"Current Mode: asm"* ]]; then
+		cat <<- EOF
+		azure CLI not in ASM mode (required for testing). Run:
+
+		  azure config mode asm
+		
+		NOTE: internal versions in PIR don't propogate to ARM stack until they're
+		published to PROD globally, hence 'asm')
+		EOF
+		exit 1
+	fi
+}
+
+intro() {
+	cat <<- EOF
+	$EXTENSION_NAME integration tests:
+
+	Execution Plan
+	==============
+	1. Create Azure VMs in the first slice region (make sure 'make publish + make replicationstatus').
+	2. Add VM extension with a config exercising features to the VMs.
+	3. Make sure the correct version has landed to the VMs.
+	4. Wait until the VMs reach the goal state provided by $EXTENSION_NAME. 
+	5. Clean up VMs.
+
+
+	EOF
+}
+
+set_subs() {
+	log "Setting subscription to $TEST_SUBS..."
+	azure account set $TEST_SUBS 1>/dev/null
+}
+
+ssh_key() {
+	echo "$SCRIPT_DIR/id_rsa"
+}
+
+ssh_pub_key() {
+	echo "$(ssh_key).pub"
+}
+
+generate_ssh_keys() {
+	local key=$(ssh_key)
+	local pub=$(ssh_pub_key)
+
+	is_file "$key" && is_file "$pub" && {
```

Comment:
if [ -f "$key" ] && [ -f "$pub" ]; then
    return
fi

It's a shell script. It should like like a shell script, not some other language.